### PR TITLE
Introduce XHGUI_CONFIG_DIR constant

### PR DESCRIPTION
**Pull request by tsari**
Introduce XHGUI_CONFIG_DIR constant as way to set the location of your xhgui config.php when using pertools/xhgui as composer library.

I did't find another solution to add xhgui (or the xhgui-collector) to an application via composer.json and getting access to an own config.php outside the vendor directory.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,7 +226,15 @@ Profile an Application or Site
 ==============================
 
 The simplest way to profile an application is to use
-`external/header.php`. `external/header.php` is designed to be combined
+`external/header.php`. 
+
+You should require `perftools/xhgui` as dependency in your projects composer.json. Add the following lines to your index.php and replace `PATH_TO_OWN_CONFIG` with the path to the directory where your xhgui `config.php` is located. Replace `PATH_TO_YOUR_VENDOR` with the path to your vendor directory.
```
Comment (markstory) [inline README.md]: I would rather we recommend people use [perftools/xhgui-collector](https://github.com/perftools/xhgui-collector) as the composer dependency in your application.

File: external/header.php [hunk 2]
```diff
@@ -67,11 +67,13 @@
...
-if (file_exists($dir . '/config/config.php')) {
-    Xhgui_Config::load($dir . '/config/config.php');
+$configDir = defined('XHGUI_CONFIG_DIR') ? XHGUI_CONFIG_DIR : $dir . '/config/';
+if (file_exists($configDir . 'config.php')) {
+    Xhgui_Config::load($configDir . 'config.php');
+} else {
+    Xhgui_Config::load($configDir . 'config.default.php');
 }
-unset($dir);
+unset($dir, $configDir);
```
Comment (markstory) [inline external/header.php]: If we add this config path constant to the collector package, I can get this code updated to use the collector. Its something that has been on my todo list for quite a while anyways.

Comment (tsari) [reply external/header.php]: Ok. I added the changes (and some more) to https://github.com/perftools/xhgui-collector and opened a PR (https://github.com/perftools/xhgui-collector/pull/3)

Comment (markstory) [reply external/header.php]: Thanks! Once that is merged I'll fix up this header file so it uses the collector but is still backwards compatible.

**Comment by markstory** [comment]
Thanks for getting the ball started on this. I decided to use this as a starting point to leverage the collector library in xhgui. The app and collector script now support the config directory constant.

**Pull request closed without merging**